packages/next/pages/_document.js

```diff
@@ -134,12 +130,22 @@ export class Head extends Component {
     return <head {...this.props}>
       {children}
       {head}
+      {amphtml && <>
+        <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1"/>
+        <link rel="canonical" href={page} />
+        <style amp-boilerplate="" dangerouslySetInnerHTML={{__html: `body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}`}}></style>
+        <noscript><style amp-boilerplate="" dangerouslySetInnerHTML={{__html: `body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}`}}></style></noscript>
+        <script async src="https://cdn.ampproject.org/v0.js"></script>
```

Comment:
suggestion: add `<link rel="preload" as="script" href="https://cdn.ampproject.org/v0.js">` (see https://www.ampproject.org/docs/fundamentals/optimize_amp#optimize-the-amp-runtime-loading).

Reply:
Done!